### PR TITLE
Task - My tasks

### PR DESCRIPTION
**Pull request by leonidbogin**
image

### CODE REVIEW

File: DIMS-Core/DIMS-Core.BusinessLayer/Models/Task/MyTaskModel.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+
+namespace DIMS_Core.BusinessLayer.Models.Task
+{
+    public class MyTaskModel
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core.BusinessLayer/Models/Task/MyTaskModel.cs:5]: Why is it `My...` ?

File: DIMS-Core/DIMS-Core/Views/Task/MyTasks.cshtml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                                <form asp-action="task-success" asp-controller="tasks" asp-route-id="@item.UserTaskId" asp-antiforgery="true">
+                                    <input type="submit" value="Success" class="btn btn-success"/>
+                                </form>
+                            }
+                            @if (item.Status != "Fail")
+                            {
+                                <form asp-action="task-fail" asp-controller="tasks" asp-route-id="@item.UserTaskId" asp-antiforgery="true">
+                                    <input type="submit" value="Fail" class="btn btn-danger"/>
+                                </form>
+                            }
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core/Views/Task/MyTasks.cshtml:97]: Please, make all buttons fixed size and add space between them

File: DIMS-Core/DIMS-Core.DataAccessLayer/Interfaces/ITaskStateRepository.cs [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 
 namespace DIMS_Core.DataAccessLayer.Interfaces
 {
     public interface ITaskStateRepository : IRepository<TaskState>
     {
-
+        void SetActive(int userId, int taskId);
+        void SetPause(int userId, int taskId);
+        void SetSuccess(int userId, int taskId);
+        void SetFail(int userId, int taskId);
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Interfaces/ITaskStateRepository.cs:10]: Why do we need 4 methods for this ? I believe you can use 1 method `SetStatus(int userId, int taskId, Status status)` for this everywhere with `Enum Status`

**Review by maximsan — CHANGES_REQUESTED** [review]
Hello! Think about using one method for statuses. It seems like a lot of redundant code for my point of view.

**Review by SashaDrapun — APPROVED** [review]
Good job!

File: DIMS-Core/DIMS-Core.BusinessLayer/Models/Task/MyTaskModel.cs (hunk 1 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.BusinessLayer/Models/Task/MyTaskModel.cs]: I did rename MyTask -> CurrentTask
```suggestion
    public class CurrentTaskModel

File: DIMS-Core/DIMS-Core.DataAccessLayer/Interfaces/ITaskStateRepository.cs (hunk 3 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Interfaces/ITaskStateRepository.cs]:
```suggestion
        void SetState(int userId, int taskId, TaskStateEnum status);

**Comment by leonidbogin** [comment]
image

File: DIMS-Core/DIMS-Core/Views/Task/MyTasks.cshtml (hunk 2 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Views/Task/MyTasks.cshtml]: image

**Review by maximsan — APPROVED** [review]
Hello! Looks better. @PushMeNow  please review.

File: DIMS-Core.Database/DIMS-Core.Database/DIMS-Core.Database.sqlproj [hunk 4]
```diff
@@ -78,6 +78,10 @@
     <Build Include="Views\vUserTrack.sql" />
     <Build Include="Procedures\DeleteUser.sql" />
     <Build Include="Functions\CalcFullAge.sql" />
+    <Build Include="Procedures\SetUserTaskAsFail.sql" />
+    <Build Include="Procedures\SetUserTaskAsSuccess.sql" />
+    <Build Include="Procedures\SetUserTaskAsPause.sql" />
+    <Build Include="Procedures\SetUserTaskAsActive.sql" />
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/DIMS-Core.Database.sqlproj]:
```suggestion
    <Build Include="Procedures\SetUserTaskState.sql" />

File: DIMS-Core/DIMS-Core/Controllers/TaskController.cs [hunk 5]
```diff
@@ -33,8 +34,29 @@ public async Task<IActionResult> Index()
...
 
+        [HttpGet("current-tasks")]
+        public async Task<IActionResult> CurrentTasks()
+        {
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home", new { });
+            }
+
+            var currentUser = memberService.GetMemberByEmail(User.Identity.Name).Result;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Controllers/TaskController.cs:45]: Use await keyword.

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 6]
```diff
@@ -16,45 +15,67 @@ public class TaskTrackController : Controller
...
         public async Task<IActionResult> Index()
         {
-            // To Do - Get the id of the current user
-            int userId = 3;
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home", new { });
+            }
+
+            var currentUser = memberService.GetMemberByEmail(User.Identity.Name).Result;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:41]: await

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 7]
```diff
@@ -16,45 +15,67 @@ public class TaskTrackController : Controller
...
+        public async Task<IActionResult> Create(int userTaskId = 0, string back = null, string backAction = null)
         {
-            // To Do - Get the id of the current user
-            int userId = 3;
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home", new { });
+            }
+
+            var currentUser = memberService.GetMemberByEmail(User.Identity.Name).Result;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:62]: await

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi dude!
I did some comments. Let me know if any questions.

File: DIMS-Core/DIMS-Core/Controllers/TaskController.cs (hunk 5 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskController.cs:45]:
```suggestion
            var currentUser = await memberService.GetMemberByEmail(User.Identity.Name);

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 6 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:41]:
```suggestion
            var currentUser = await memberService.GetMemberByEmail(User.Identity.Name);

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 7 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:62]:
```suggestion
            var currentUser = await memberService.GetMemberByEmail(User.Identity.Name);

**Pull request merged**